chef/cookbooks/cpe_chrome/resources/cpe_chrome.rb

```diff
@@ -27,7 +27,8 @@ def manage_chrome
   end
 
   def manage_chrome_osx
-    prefs = node['cpe_chrome'].reject { |_k, v| v.nil? }
+    prefs = node['cpe_chrome']['profile'].reject \
+      { |_k, v| v.nil? || v.class == Chef::Node::ImmutableArray && v.empty? }
```

Comment:
I will test these, but I think some of these suggestions may have other unintended consequences.

As for the ImmutableArray, it is exactly what we want in this case.